polygeist/test/polygeist-opt/sycl/loop_internalization.mlir

```diff
@@ -158,8 +160,11 @@ gpu.func @kernel(%arg0: memref<?x!sycl_accessor_2_f32_r_gb>, %arg1: memref<?x!sy
 // CHECK-DAG:           [[IV1_CAST:%.*]] = arith.index_cast [[IV1]] : index to i64   
 // CHECK-DAG:           [[IV3_CAST:%.*]] = arith.index_cast [[IV3]] : index to i64 
 // CHECK-NEXT:          sycl.constructor @id([[ID:%.*]], [[TX]], [[IV1_CAST]], [[IV3_CAST]]) {{.*}} : (memref<?x!sycl_id_3_>, i64, i64, i64)
-// CHECK-NEXT:          [[SUBSCR:%.*]] = sycl.accessor.subscript %arg0[[[ID]]] : (memref<?x!sycl_accessor_3_f32_r_gb>, memref<?x!sycl_id_3_>) -> memref<?xf32>
-// CHECK-NEXT:          {{.*}} = affine.load [[SUBSCR]][0] : memref<?xf32>
+// CHECK-NEXT:          [[SUBSCR1:%.*]] = sycl.accessor.subscript %arg0[[[ID]]] : (memref<?x!sycl_accessor_3_f32_r_gb>, memref<?x!sycl_id_3_>) -> memref<?xf32>
+// CHECK-NEXT:          {{.*}} = affine.load [[SUBSCR1]][0] : memref<?xf32>
+// CHECK-NEXT:          sycl.constructor @id([[ID:%.*]], [[TX]], [[TY]], [[TZ]]) {{.*}} : (memref<?x!sycl_id_3_>, i64, i64, i64)
+// CHECK-NEXT:          [[SUBSCR2:%.*]] = sycl.accessor.subscript %arg0[[[ID]]] : (memref<?x!sycl_accessor_3_f32_r_gb>, memref<?x!sycl_id_3_>) -> memref<?xf32>
+// CHECK-NEXT:          {{.*}} = affine.load [[SUBSCR2]][0] : memref<?xf32>
```

Comment:
The array accesses aren't yet changed to use local memory, the local memory is only allocated (@whitneywhtsang is working on replacing the references the analysis indicates they should go to shared local memory). That is the reason the first load (the one that uses SUBSCR1 does not yet use shared local memory). However the check at line 156 shows that the compiler create a `memref.view` operation to get a pointer to local memory for the first array access.

In this PR I have added a load that is perfectly coalesced and so should *not* use shared local memory. If the analysis incorrectly classified as a access that should go to shared local memory, then the compiler would have generated 2 memref.view operations. Because it did not we can infer that the compiler correctly classified it as a coalesced access. 

This is a bit circuitous, I know, however once we get the latest PRs from @whitneywhtsang work in, it will be clearer the transformation did the right thing.